lib/lambda_ethereum_consensus/p2p/block_downloader.ex

```diff
@@ -0,0 +1,132 @@
+defmodule LambdaEthereumConsensus.P2P.BlockDownloader do
+  @moduledoc """
+  This module requests blocks from peers.
+  """
+  alias LambdaEthereumConsensus.P2P
+  use GenStage
+
+  @protocol_id "/eth2/beacon_chain/req/beacon_blocks_by_range/2/ssz_snappy"
+
+  @impl true
+  def init([host]) do
+    {:producer, host}
+  end
+
+  @impl true
+  def handle_demand(incoming_demand, host) do
+    blocks =
+      for _ <- 0..incoming_demand do
+        request_block(host) |> wrap_message()
+      end
+
+    {:noreply, blocks, host}
+  end
+
+  defp request_block(host) do
+    # TODO: get missing slots from DB
+    start_slot = 7_270_097 + :rand.uniform(100_000)
+
+    # TODO: handle no-peers asynchronously?
+    peer_id = get_some_peer()
+
+    payload =
+      %SszTypes.BeaconBlocksByRangeRequest{
+        start_slot: start_slot,
+        # TODO: we need to refactor the Snappy library to return
+        # the remaining buffer when decompressing
+        count: 1
+      }
+
+    # This should never fail
+    {:ok, encoded_payload} = payload |> Ssz.to_ssz()
+
+    size_header =
+      encoded_payload
+      |> byte_size()
+      |> P2P.Utils.encode_varint()
+
+    # This should never fail
+    {:ok, compressed_payload} = encoded_payload |> Snappy.compress()
+
+    with {:ok, stream} <- Libp2p.host_new_stream(host, peer_id, @protocol_id),
+         :ok <- Libp2p.stream_write(stream, size_header <> compressed_payload),
+         :ok <- Libp2p.stream_close_write(stream),
+         {:ok, chunk} <- read_response(stream),
+         {:ok, block} <- decode_response(chunk) do
+      block
+    else
+      # we just ignore the error and continue
```

Comment:
Let's also log the error. If it's too verbose, we can add it as `debug` level for now.